includes/blocks/course-list/class-sensei-course-list-block.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * File containing extra functionalities of extended Sensei_Course_List_Block class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Course_List_Block
+ */
+class Sensei_Course_List_Block {
+
+	/**
+	 * Sensei_Course_List_Block constructor.
+	 */
+	public function __construct() {
+		add_filter( 'render_block', [ $this, 'render_login_form_if_applicable' ], 10, 2 );
+	}
+
+	/**
+	 * Replaces content of Course List block when Show Login Form is set and user is logged out.
+	 *
+	 * @param string $block_content The block content to be rendered.
+	 * @param array  $block          The block to be rendered.
+	 *
+	 * @return string
+	 */
+	public function render_login_form_if_applicable( $block_content, $block ) {
+		if ( 'core/query' !== $block['blockName'] ) {
+			return $block_content;
+		}
+
+		global $post;
+
+		$is_course_list_block = array_key_exists( 'query', $block['attrs'] ) &&
+			'course' === $block['attrs']['query']['postType'] &&
+			array_key_exists( 'className', $block['attrs'] ) &&
+			false !== strpos( $block['attrs']['className'], 'wp-block-sensei-lms-course-list' );
+
+		$is_my_courses_page = isset( $post ) && is_page() && intval( Sensei()->settings->get( 'my_course_page' ) ) === $post->ID;
```

Comment:
Do you think the following check might be enough and a bit easier to read? You could also clean up the global in this case.

```suggestion
		$is_my_courses_page = get_the_ID() === (int) Sensei()->settings->get( 'my_course_page' );
```

Reply:
Thanks for the suggestion, updated it here https://github.com/Automattic/sensei/pull/6343/commits/b17e2c624a071fb7b721d18cc21ead880430d44f

Reply:
Nice! One small thing - you can also clean up the `global $post;` located a few lines up.

Reply:
Oh, just noticed that it got left out from my commit, removed [here](https://github.com/Automattic/sensei/pull/6343/commits/294148e85e03a340f3f7507e0866c16a25caedfe)